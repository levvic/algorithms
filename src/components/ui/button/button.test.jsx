import React from "react";
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

describe("Check props button", () => {
    it("Text on button is rendered correctly", () => {
        const tree = renderer.create(<Button text={"Test"} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("Button without text", () => {
        const tree = renderer.create(<Button />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("Button is disabled", () => {
        const tree = renderer.create(<Button disabled />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("Button shows isLoading icon", () => {
        const tree = renderer.create(<Button isLoader />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Button onClick callback functionality test', () => {

        window.alert = jest.fn();

        render(<Button onClick={() => window.alert('text')} text={'test text'} />)

        const button = screen.getByText('test text');
        fireEvent.click(button);
        expect(window.alert).toHaveBeenCalledWith('text');
    })
});