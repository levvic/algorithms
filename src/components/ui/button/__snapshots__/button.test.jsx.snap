// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check props button Button is disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Check props button Button shows isLoading icon 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Check props button Button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Check props button Text on button is rendered correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Test
  </p>
</button>
`;
